torch/csrc/jit/passes/onnx/function_extraction.cpp

```diff
@@ -0,0 +1,1098 @@
+      env[n->output(i)] = clone_n->output(i);
+    }
+    g->insertNode(clone_n);
+  }
+
+  // If values are used outside of this graph, set as graph output.
+  for (auto* v : ctx.outputs_) {
+    TORCH_INTERNAL_ASSERT(env.find(v) != env.end());
+    g->registerOutput(env[v]);
+  }
+
+  GRAPH_DEBUG(g->toString());
+  return g;
+}
+
+Node* FunctionExtractor::CreateFunctionDefNode(
+    FunctionContext& func_ctx,
+    const std::shared_ptr<Graph>& graph,
+    const std::string& domain_name,
+    const std::string& func_name) {
+  const auto func_def_nk = Symbol::onnx("LocalFunctionDef");
+  const auto func_g_attr = Symbol::attr("graph");
+  const auto func_name_attr = attr::name;
+  const auto func_domain_attr = Symbol::attr("domain");
+
+  auto func_graph = ConstructFuncGraph(func_ctx);
+
+  // create and insert local function definition node
+  auto func_def_n = graph->create(func_def_nk, 0);
+  func_def_n->g_(func_g_attr, func_graph);
+  func_def_n->s_(func_name_attr, func_name);
+  func_def_n->s_(func_domain_attr, domain_name);
+  graph->prependNode(func_def_n);
+
+  // set constants and attributes of different values as function attributes.
+  std::unordered_map<std::string, int> base_attr_name_count;
+  std::vector<std::string> final_attr_names;
+
+  auto adjust_attr_name = [&](std::string attr_name) {
+    if (base_attr_name_count.find(attr_name) != base_attr_name_count.end()) {
+      attr_name =
+          attr_name + "." + std::to_string(base_attr_name_count[attr_name]++);
+    } else {
+      base_attr_name_count[attr_name] = 1;
+    }
+    return attr_name;
+  };
+
+  for (const auto c_it : func_ctx.constant_map_) {
+    auto* c_n = c_it.first;
+    TORCH_INTERNAL_ASSERT(c_n->kind() == c10::onnx::Constant);
+    auto attr_name = std::string(c_n->kind().toUnqualString()) + '_' +
+        c_n->output()->debugNameBase();
+    auto final_attr_name = adjust_attr_name(attr_name);
+    final_attr_names.emplace_back(final_attr_name);
+    func_ctx.SetAttrName(c_n, final_attr_name);
+  }
+  for (const auto n_it : func_ctx.attribute_map_) {
+    auto* n = n_it.first;
+    for (const auto attr_it : n_it.second) {
+      const auto& attr = attr_it.first;
+      auto attr_name =
+          std::string(n->kind().toUnqualString()) + '_' + attr.toUnqualString();
+      auto final_attr_name = adjust_attr_name(attr_name);
+      final_attr_names.emplace_back(final_attr_name);
+      func_ctx.SetAttrName(n, attr, final_attr_name);
+    }
+  }
+
+  func_def_n->ss_(Symbol::attr("attributes"), final_attr_names);
+
+  return func_def_n;
+}
+
+Node* FunctionExtractor::CreateFunctionNode(
+    FunctionContext& func_ctx,
+    ScopeContext& scope_ctx,
+    const std::shared_ptr<Graph>& graph,
+    const std::string& domain_name,
+    const std::string& func_name) {
+  const auto& func_scope = func_ctx.scope_key_;
+  const auto& func_scope_ctx = func_ctx.scope_ctxs_[func_scope];
+  const auto func_nk = Symbol::fromQualString(domain_name + "::" + func_name);
```

Comment:
I find this name `func_nk` a little confusing, and it's only used once. I suggest just in-lining the function call into line 562 and get rid of the local variable.